src/CommunityToolkit.Maui/Views/Popup/PopupExtensions.shared.cs

```diff
@@ -43,13 +52,34 @@ public static partial class PopupExtensions
 	/// </returns>
 	public static Task<object?> ShowPopupAsync<TPopup>(this Page page, TPopup popup) where TPopup : Popup
 	{
-#if WINDOWS
-		return PlatformShowPopupAsync(popup, GetMauiContext(page));
-#else
+		if (page.IsPlatformEnabled)
+		{
+			return CreateAndShowPopupAsync(page, popup);
+		}
+		else
+		{
+			var taskCompletionSource = new TaskCompletionSource<object?>();
 
-		CreatePopup(page, popup);
-		return popup.Result;
-#endif
+			async void handler(object? sender, NavigatedToEventArgs args)
+			{
+				page.NavigatedTo -= handler;
+
+				try
```

Comment:
This will be the easiest way to handle the exception. The `ContinueWith` will just run if the exception happens inside the `popup.Result` `Task` otherwise it will crash. So changing this local function to be an `async void` and wrapping the important pieces on a `try/catch block`

Reply:
Thanks for checking this. Yeah I couldn't find a way to do it without using that magic `await` word.

Reply:
No worries, exceptions are odd when handling `Task`s I forgot about it.